### PR TITLE
Add nil check to interface test

### PR DESCRIPTION
**Pull request by robert-w-gries**
Add nil check to prevent stack trace when `capabilities` on switch cannot be found

### CODE REVIEW

File: tests/test_interface.rb [hunk 1]
```diff
@@ -141,15 +141,15 @@ def interface_count
   def capable_speed_values(interface)
     capabilities = config("show interface #{interface.name} capabilities")
     speed_capa = capabilities.match(/Speed:\s+(\S+)/)
-    return [] if speed_capa[1].nil?
+    return [] if speed_capa.nil? || speed_capa[1].nil?
```
Comment (mikewiebe) [inline tests/test_interface.rb:144]: I think it might be helpful to log a warning when speed and duplex capabilities cannot be detected.

Comment (robert-w-gries) [inline tests/test_interface.rb:144]: Done

File: tests/test_interface.rb [hunk 2]
```diff
@@ -576,6 +576,7 @@ def test_speed
 
     # Test up to two non-default values
     speed_values = capable_speed_values(interface)
+    warn('No compatible speeds found on interface') if speed_values.nil?
```
Comment (mikewiebe) [inline tests/test_interface.rb]: **Nit** Include the interface name.. otherwise :+1:

Comment (robert-w-gries) [inline tests/test_interface.rb]: Done.  I also corrected the method to be `empty?` instead of `nil?`

**Pull request merged**